### PR TITLE
Fix warnings that occur during PHP testing with coverage

### PR DESCRIPTION
**Pull request by pierlon**
## Summary

This PR fixes the [several warnings](https://travis-ci.org/github/ampproject/amp-wp/jobs/717430538#L847) that occur while running PHP tests with coverage. The warnings are mostly caused by incorrect uses of the `@covers` and `@uses` annotations, which require a FQCN either on the test class or method. 

## Checklist

- [ ] My pull request is addressing an open issue (please create one otherwise).
- [ ] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

**Comment by pierlon** [comment]
@westonruter Any reason why the PHP testing with coverage job is limited to the `develop` branch?

**Comment by westonruter** [comment]
@pierlon No good reason I can recall. May have been an effort to reduce builds or thinking that if a PR is not targeting develop that it doesn't need linting yet. Go ahead and remove that restriction.

File: .travis.yml [hunk 1]
```diff
@@ -145,7 +145,6 @@ jobs:
       env: WP_VERSION=trunk  DEV_LIB_ONLY=phpunit                               INSTALL_PWA_PLUGIN=1
 
     - name: PHP and JavaScript unit tests (7.4, WordPress trunk, with code coverage)
-      if: branch = develop AND type = push
```
Comment (westonruter) [inline .travis.yml]: I think this was put in place because otherwise the PR builds take very long, no?

**Review by westonruter — CHANGES_REQUESTED** [review]
Actually, I recall now why the `if` condition was there for unit tests: the build takes takes much longer when coverage is gathered. So by limiting it to `develop` then it doesn't get in the way of PRs. This was the reasoning when coverage was first introduced anyway, and it may not be a valid reason anymore. 

Build is failing now for `Test_AMP_Validated_URL_Post_Type::test_delete_stylesheets_postmeta_batch()` when coverage is reported: https://travis-ci.org/github/ampproject/amp-wp/jobs/717628370#L851-L863

**Comment by westonruter** [comment]
Strange that phpunit is failing in `PHP and JavaScript unit tests (7.4, WordPress trunk, with code coverage)` but the build is passing. 😕

**Comment by pierlon** [comment]
That's because it's allowed to fail, since it's testing against WP trunk.

File: .travis.yml (hunk 1 above)
Comment (pierlon) [reply .travis.yml]: Yea that makes sense, would be great to have coverage for each PR nonetheless.

**Comment by westonruter** [comment]
There's two jobs for `trunk`:

1. `PHP unit tests (8.0, WordPress trunk)` has a build error
2. `PHP and JavaScript unit tests (7.4, WordPress trunk, with code coverage)` has a build failure

I thought only the first would be allowed to fail because it is listed among `allow_failures`. At least, according to “These need to be exact matches, including whitespace!” Maybe that's no longer true.

**Comment by pierlon** [comment]
The coverage job is allowed to fail since it matches:

https://github.com/ampproject/amp-wp/blob/aed44dddee5890fcd3743a5a4b5d1c1d96f74127/.travis.yml#L64

I just realized it's not actually using `trunk`, but `latest` :thinking:.

**Comment by pierlon** [comment]
> Build is failing now for Test_AMP_Validated_URL_Post_Type::test_delete_stylesheets_postmeta_batch() when coverage is reported: https://travis-ci.org/github/ampproject/amp-wp/jobs/717628370#L851-L863

Fixed in https://github.com/ampproject/amp-wp/pull/5224/commits/c445e52348d97f8b0a251c41725c7befd9e23b5f (see https://travis-ci.org/github/ampproject/amp-wp/jobs/718309089).

**Review by westonruter — APPROVED** [review]

**Pull request merged**